src/Language/Haskell/LSP/Core.hs

```diff
@@ -846,7 +847,7 @@ initializeRequestHandler' onStartup mHandler tvarCtx req@(J.RequestMessage _ ori
               , J._documentHighlightProvider        = supported (documentHighlightHandler h)
               , J._documentSymbolProvider           = supported (documentSymbolHandler h)
               , J._workspaceSymbolProvider          = supported (workspaceSymbolHandler h)
-              , J._codeActionProvider               = supported (codeActionHandler h)
+              , J._codeActionProvider               = codeActionProvider o
```

Comment:
This line of the patch is incorrect. It should be implemented like `typeDefinitionProvider`.